snap/pack/pack.go

```diff
@@ -227,33 +212,92 @@ func Snap(sourceDir string, opts *Options) (string, error) {
 		return "", fmt.Errorf("cannot use compression %q", opts.Compression)
 	}
 
-	info, err := prepare(sourceDir, opts.TargetDir)
+	// ensure we have valid content
+	packFunc := packSnap
+	yaml, err := os.ReadFile(filepath.Join(sourceDir, "meta", "snap.yaml"))
+	if err != nil {
```

Comment:
should we check specifically for os.IsNotExist, otherwise we wil always turn the error into one about compoenent.yaml?